components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import { ArrowLeft, ArrowRight } from 'react-feather';
+
+import { PaginationProps } from '@/components/Pagination/Pagination.types';
+
+const Pagination = ({
+  current,
+  total,
+  onPageClick,
+  limit = 1
+}: PaginationProps) => {
+  const [isPrevAvailable, setIsPrevAvailable] = useState(false);
+  const [isNextAvailable, setIsNextAvailable] = useState(false);
+  const [pages, setPages] = useState<number[]>([current]);
+  const [centerPage, setCenterPage] = useState(0);
+
+  const calculatePages = (
```

Comment:
this should be a util function.
+
there is options in this function like minPage, maxPages which maybe should be in Pagination logic as a props...

Reply:
I'm adding it to the to-do list.